### PR TITLE
Use std::lock_guard instead of std::unique_lock.

### PR DESCRIPTION
**Pull request by toregge**
@geirst, please review.
@baldersheim, FYI

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/server/pendinglidtracker.cpp [hunk 1]
```diff
@@ -38,7 +37,7 @@ PendingLidTracker::consume(uint32_t lid) {
 
 void
 PendingLidTracker::waitForConsumedLid(uint32_t lid) {
-    LockGuard guard(_mutex);
+    std::unique_lock<std::mutex> guard(_mutex);
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/pendinglidtracker.cpp]: Why are you still using unique_lock here?

**Comment by geirst** [comment]
Overall looks good, please have a look at my question

File: searchcore/src/vespa/searchcore/proton/server/pendinglidtracker.cpp (hunk 1 above)
Comment (toregge) [reply searchcore/src/vespa/searchcore/proton/server/pendinglidtracker.cpp]: Because we need a std::unique_lock when waiting on a std::condition_variable.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**